lib/omni_auth/strategies/abstract_saml.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require 'omniauth'
+require 'omniauth-saml'
+require 'ruby-saml'
+
+module OmniAuth
+  module Strategies
+    class AbstractSaml < OmniAuth::Strategies::SAML
+      # We want to specify some security options ourselves
+      option :security, {
+        digest_method: XMLSecurity::Document::SHA512,
+        signature_method: XMLSecurity::Document::RSA_SHA512,
+        metadata_signed: true, # Enable signature on Metadata
+        authn_requests_signed: true, # Enable signature on AuthNRequest
+        logout_requests_signed: true, # Enable signature on Logout Request
+        logout_responses_signed: true, # Enable signature on Logout Response
+        want_assertions_signed: true, # Require the IdP to sign its SAML Assertions
+        want_assertions_encrypted: true # Invalidate SAML messages without an EncryptedAssertion
+      }
+
+      # Our auto-login mechanism passes a desired redirect path (signed with a
+      # secret to prevent tampering) to the request phase.
+      # If we forward this via SAML's standard "RelayState" parameter, we will
+      # get it back in the callback phase.
+      # TODO: This parameter is not yet used in the app
+      option :idp_sso_target_url_runtime_params, {
+        redirect_path: 'RelayState'
+      }
+
+      # This Lambda is responsible for terminating the session and will
+      # only be used for identity-provider-initiated logout requests
+      option :idp_slo_session_destroy, (proc { |env, _session|
```

Comment:
The session destroy is tricky and I am not completely satisfied yet. This mechanism works (as tested) but has limitations due to cookie usage. An ideal solution would use another session store in `config/initializers/session_store.rb`. Any input here is appreciated.

Reply:
Let's discuss this in person.

Reply:
Let's monitor this in production and create a ticket for it.

Reply:
I created #589 for that